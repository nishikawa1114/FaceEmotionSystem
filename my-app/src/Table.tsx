import React from 'react';
import './index.css';
import { Emotion } from './types';


interface TableProps {
    emotion: Emotion;
}

export class Table extends React.Component<TableProps> {

    render() {

        return (
            <div>
                <table key="emotion_table">
                    <tr key="table_header">
                        <th>感情</th>
                        <th>値</th>
                    </tr>
                    {Object.entries(this.props.emotion).map(([key, value]) => (

                        <tr>
                            <td className="emotion">{key}</td>
                            <td className="emotion_value">{value}</td>
                        </tr>

                    ))}
                </table>
            </div>
        )
    }
}