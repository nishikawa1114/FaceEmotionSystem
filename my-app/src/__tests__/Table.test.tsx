import { Table } from "../Table";
import { create } from "react-test-renderer";
import React from "react";

interface Emotion {
    anger: number;
    contempt: number;
    disgust: number;
    fear: number;
    happiness: number;
    neutral: number;
    sadness: number;
    surprise: number;
}

const emotions: Emotion = {
    anger: 0.1,
    contempt: 0.2,
    disgust: 0.1,
    fear: 0,
    happiness: 0.005,
    neutral: 0.4,
    sadness: 0.01,
    surprise: 0.002,
}
// const img: Image = { id: 1, url: "https://nishikawa.blob.core.windows.net/images/steve/2020/10/15/01.jpg?sv=2019-07-07&sr=c&si=myPolicyPS&sig=FkKJ4nXCiqzDYjbSaDfqli%2FnErPRTKrD%2BUQfH0MT3ac%3D" };

it("ImageInfo test", () => {
    const tree = create(
        <Table
            emotion={emotions}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();

})