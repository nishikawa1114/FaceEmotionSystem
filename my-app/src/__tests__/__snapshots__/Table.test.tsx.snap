// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageInfo test 1`] = `
<div>
  <table>
    <tr>
      <th>
        感情
      </th>
      <th>
        値
      </th>
    </tr>
    <tr>
      <td
        className="emotion"
      >
        anger
      </td>
      <td
        className="emotion_value"
      >
        0.1
      </td>
    </tr>
    <tr>
      <td
        className="emotion"
      >
        contempt
      </td>
      <td
        className="emotion_value"
      >
        0.2
      </td>
    </tr>
    <tr>
      <td
        className="emotion"
      >
        disgust
      </td>
      <td
        className="emotion_value"
      >
        0.1
      </td>
    </tr>
    <tr>
      <td
        className="emotion"
      >
        fear
      </td>
      <td
        className="emotion_value"
      >
        0
      </td>
    </tr>
    <tr>
      <td
        className="emotion"
      >
        happiness
      </td>
      <td
        className="emotion_value"
      >
        0.005
      </td>
    </tr>
    <tr>
      <td
        className="emotion"
      >
        neutral
      </td>
      <td
        className="emotion_value"
      >
        0.4
      </td>
    </tr>
    <tr>
      <td
        className="emotion"
      >
        sadness
      </td>
      <td
        className="emotion_value"
      >
        0.01
      </td>
    </tr>
    <tr>
      <td
        className="emotion"
      >
        surprise
      </td>
      <td
        className="emotion_value"
      >
        0.002
      </td>
    </tr>
  </table>
</div>
`;
